app/jobs/bill_recurring_pay_in_advance_charges_job.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class BillRecurringPayInAdvanceChargesJob < ApplicationJob
+  queue_as 'billing'
+
+  def perform(subscriptions, timestamp)
+    Fees::CreateRecurringPayInAdvanceService.call(subscriptions:, billing_at: Time.zone.at(timestamp))
+
+    # TODO: Should we retry the job if it fails?
+    # result.raise_if_error!
```

Comment:
the raise_if_error is here to make sure we are not silencing any problems. Since the service is built with the `billing_at` input, and should use a DB transaction, I guess we are sage to raise and retry if mandatory.
What do you think?

Reply:
since the service enqueues a job and verifies if we have the data needed, does retrying make sense?